core/include/kernel/tee_ta_manager.h

```diff
@@ -95,7 +95,7 @@ struct tee_ta_session {
 	bool cancel;		/* True if TAF is cancelled */
 	bool cancel_mask;	/* True if cancel is masked */
 	TEE_Time cancel_time;	/* Time when to cancel the TAF */
-	void *user_ctx;		/* ??? */
+	void *user_ctx;		/* Client TA context or NULL (pTA invoked) */
```

Comment:
I'd rather describe it as "Opaque session handle, assigned by PTA".
The only usage I found so far is in `core/arch/arm/tee/pta_socket.c` and that could be replaced by other means.
It seems this field could be made redundant with not too much effort.

Reply:
Then maybe better remove this field and update `pta_socket.c` accordingly. Ok?

Reply:
Or we could do both, adding an accurate comment here is an improvement regardless.

Reply:
I'll wait https://github.com/OP-TEE/optee_os/pull/2815/commits/f8f9dc947f5508d6c4140b3930cf4efce981f0f0 is merged before proposing an update.

Reply:
Ok, I'll update this P-R according to your comment and postpone `user_ctx` field removal to a later P-R.